### PR TITLE
Access control: Add descriptions to fixed roles

### PR DESCRIPTION
**Pull request by vtorosyan**
**What this PR does / why we need it**:

Predefined roles should have proper descriptions as users are getting those back from the API and they would need information to understand the purpose etc.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by sakjur** [comment]
Implementation looks good, but I think @osg-grafana will want to work with making sure these descriptions are good 🙂

(they'll eventually show up in the UI as a human friendly description of a predefined role)

**Review by osg-grafana — CHANGES_REQUESTED** [review]
Blocking until a review can be done. Cannot go out as is.

**Comment by vtorosyan** [comment]
@mjseaman @osg-grafana do we prefer to merge this before beta2, or wait for the main release? Won't be end of the world if we don't merge this, roles won't have descriptions returned when used from the API, but if we ship initial version of docs, descriptions will be in the docs at least.

**Comment by vtorosyan** [comment]
@donomii would appreciate if you can take this one also as part of docs, since this adds descriptions for roles from the documentation.

File: pkg/services/accesscontrol/roles.go
Comment (osg-grafana) [inline pkg/services/accesscontrol/roles.go:8]:
```suggestion
	Description: "Allows to read LDAP and status information.",
```
Allows [what? / you?]

File: pkg/services/accesscontrol/roles.go
Comment (osg-grafana) [inline pkg/services/accesscontrol/roles.go:22]:
```suggestion
	Description: "Allows every read action for LDAP and in addition allows to administer LDAP.",
```
Allows every read action [to do what?] ... allows [who?]

File: pkg/services/accesscontrol/roles.go
Comment (osg-grafana) [inline pkg/services/accesscontrol/roles.go:33]:
```suggestion
	Description: "Allows to get user organizations.",
```
Allows [who?]

File: pkg/services/accesscontrol/roles.go
Comment (osg-grafana) [inline pkg/services/accesscontrol/roles.go:45]:
```suggestion
	Description: "Allows every read action for user organizations and in addition allows to administer user organizations.",
```
Allows [what? to? for?] every read action ... allows [what? / who?] to administer user organizations.

File: pkg/services/accesscontrol/roles.go
Comment (osg-grafana) [inline pkg/services/accesscontrol/roles.go:65]:
```suggestion
	Description: "Allows to list and get users and related information.",
```
Allows [what? / who?] to list...

File: pkg/services/accesscontrol/roles.go
Comment (osg-grafana) [inline pkg/services/accesscontrol/roles.go:89]:
```suggestion
	Description: "Allows every read action for users and in addition allows to administer users.",
```
Allows [what? who ... to?] ... allows [what? who?]

File: pkg/services/accesscontrol/roles.go
Comment (osg-grafana) [inline pkg/services/accesscontrol/roles.go:136]:
```suggestion
	Description: "Allows to reload provisioning configurations.",
```
Allows [who? / what?] to reload ...

**Review by osg-grafana — CHANGES_REQUESTED** [review]

File: pkg/services/accesscontrol/roles.go
Comment (gamab) [reply pkg/services/accesscontrol/roles.go:8]: Thanks @osg-grafana for reviewing. I'm no expert here, but I'll try to be useful :slightly_smiling_face: 

Side note: A role is a list of permissions that can be granted to: User(s), Team(s), Builtin Role(s) ("Viewer", "Editor", "Admin", "Grafana Admin"). In the end, thanks to an inheritance mechanism, it always allow an end user to perform an "action" (often on a "scope").
Here we have two actions from what I see, LDAPUsersRead (which most probably protects the http endpoint to list ldap users) and ActionLDAPStatusRead (which most probably protects the http endpoint to get the connection status of Grafana with the LDAP server(s))

With that in mind, do you think this description is clearer:
```suggestion
	Description: "Role to allow users to read LDAP information and get LDAP connection status.",
```
However, since it's a role description,  `"Role to allow users to"` seems a bit redundant, I think that's why the description is currently `"Allows to..."`

File: pkg/services/accesscontrol/roles.go
Comment (gamab) [reply pkg/services/accesscontrol/roles.go:22]: This role seem to inherit its permission list from the ldapAdminReadRole role (so ActionLdapUsersRead and ActionLDAPStatusRead) and on top of that there is a new permission appended to the list: ActionLDAPUsersSync. I'm assuming this protect the HTTP endpoint to retrieve/update users from an ldap DIT and remove users when they are not in the ldap DIT.
The current description states that the role grants the ldapAdminReadRole's actions with "every read action for LDAP" and on top of that, it grants permissions to "administer LDAP", I suppose this is intentionally large to encompass future administering actions.
This seems to specific:
```suggestion
	Description: "Role to allow users everything ldapAdminReadRole allows and synchronize users with an LDAP server.",
```

File: pkg/services/accesscontrol/roles.go
Comment (gamab) [reply pkg/services/accesscontrol/roles.go:33]:
```suggestion
	Description: "Role to allow users to list users in a given organization.",
```

File: pkg/services/accesscontrol/roles.go
Comment (gamab) [reply pkg/services/accesscontrol/roles.go:136]: This I know :smile:, this is mine
We have an endpoint that triggers the reloading (re-read, re-apply) of the provisioning configuration files.
I think the description is spot on.
If we really have to change, maybe:
```suggestion
	Description: "Role to allow users to reload the provisioning configuration files.",
```

File: pkg/services/accesscontrol/roles.go
Comment (gamab) [reply pkg/services/accesscontrol/roles.go:45]: Again, I'm trying to propose something, but I like the "administer" word being large enough to encompass future changes:
```suggestion
	Description: "Role to allow users everything usersOrgReadRole allows and to add/update/remove users from an organization.",
```

File: pkg/services/accesscontrol/roles.go
Comment (gamab) [reply pkg/services/accesscontrol/roles.go:65]:
```suggestion
	Description: "Role to allow users to list users in a given organization, retrieve the teams they are part of, list their authentication tokens, get their connection quotas.",
```

**Review by gamab — COMMENTED** [review]
I tried to propose an alternative to the descriptions to hopefully help in the decision making. I personally prefer the roles description as they are now. I'm not sure the usersOrgEditRole description is correct as it is.

File: pkg/services/accesscontrol/roles.go
Comment (donomii) [reply pkg/services/accesscontrol/roles.go:8]: I'd remove the first part of the sentence entirely.  We don't want to say who is doing it, because we may be adding more in the future (e.g. service accounts).  Unless you want something like "Allows client" or "Allows supplicant"

We can put something in the column header like "These permissions allow a client to..."

Comment (donomii) [inline pkg/services/accesscontrol/roles.go:8]:
```suggestion
	Description: "Read LDAP information and status.",
```

**Review by osg-grafana — APPROVED** [review]
Approving to unblock, so we can get this in as-is in the next 15 minutes. We can come back to this and fix it after the deadline passes.

**Review by osg-grafana — APPROVED** [review]

**Pull request closed without merging**
